monai/utils/safeeval.py

```
@@ -0,0 +1,73 @@
+    Returns:
+        The evaluated expression value, using `eval` with `globals_vars` and `locals_vars`
+    """
+    parsed = ast.parse(expr.strip(), mode="eval")
+
+    # collect nodes in the AST which aren't permitted and unparse them for inclusion in the exception message
+    disallowed = [ast.unparse(n) for n in ast.walk(parsed) if not isinstance(n, tuple(allowed_types))]
+
+    if disallowed:
+        raise ValueError(f"Unsafe expression `{expr}` not evaluated, contains disallowed components: {disallowed}")
+
+    return eval(expr, dict(globals_vars) if globals_vars else None, locals_vars)
```

Ruff (0.15.17)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)